flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/FineGrainedSlotManager.java

```diff
@@ -485,6 +497,16 @@ private void checkResourceRequirements() {
         }
     }
 
+    private void triggerCheckResourceRequirements() {
```

Comment:
I would suggest to call this `checkResourceRequirementsWithDelay`, and explain the purpose in comments.

Reply:
And maybe comment for `checkResourceRequirements` that is should only be called from `checkResourceRequirementsWithDelay`.